Add combine history to coadd2d FITS headers; load header into SpecUtils

tbowers7 opened the pull request:
### Description

This PR contains two small metadata _output_ additions.  No changes made to data containers or other PypeIt internals.

* Minor addition to ``coadd2d`` outputs:  Add the list of combined files to the ``spec2d`` and ``spec1d`` outputs.
* Add the primary FITS header (not all of the extensions) to the ``specutils.Spectrum`` objects in the PypeIt loader, following the convention outlined in the [SpecUtils docs](https://specutils.readthedocs.io/en/stable/spectrum.html#metadata).

### Rationale

``Calibrations/`` files contain information about what raw files were combined to produce the output; this extends the functionality to ``coadd2d``.  For the PypeIt ``specutils.Spectrum`` loaders, it is helpful to have metadata about the spectrum available for downstream processing; adding the primary header adds minimal overhead to the ``specutils.Spectrum`` object and aligns with SpecUtils practice.

### Late-Additions

Also including a logged message when a good standard star is found for ``sensfunc`` computation (and not just throwing ERRORs when an object is not found in an archive).

Also also tweaking ``pypeit_setup_coadd2d`` to remove spaces in object names (when used or searching for filenames) and only produce ``.coadd2d`` files for objects that include either 'science' or 'standard' in the frametype.

### DevSuite

While there are no expected issues with tests, etc., I am running the DevSuite to be sure (``tbowers-kst-dec01-p4rks``).

tbowers7 committed:
Include FITS header in specutils Spectrum loader

Following the convention mentioned in the SpecUtils documentation at
https://specutils.readthedocs.io/en/stable/spectrum.html#metadata
include the ``spec1d`` FITS header in the metadata for the Spectrum
object loaded from PypeIt.

	modified:   pypeit/specutils/pypeit_loaders.py

tbowers7 committed:
Add file history to coadd_2dspec

	modified:   pypeit/history.py
	modified:   pypeit/scripts/coadd_2dspec.py

tbowers7 committed:
Add more metadata

	modified:   pypeit/scripts/coadd_2dspec.py

debora-pe commented on pypeit/history.py:
```diff
@@ -163,6 +163,29 @@ def add_coadd1d(self, spec1d_files, objids, gpm_exp=None):
+            objname (:obj:`str`): Name of the object being coadded
+        """
+        # Add history
+        self.append(f'PypeIt Coadding target {objname} in '
```
Maybe use Coadded here?

debora-pe commented on pypeit/scripts/coadd_2dspec.py:
```diff
@@ -223,12 +224,17 @@ def main(args):
+
         # THE FOLLOWING MIMICS THE CODE IN pypeit.save_exposure()
         subheader = spectrograph.subheader_for_spec(head2d, head2d)
+        subheader['coaddobj'] = (basename, 'Coadd object base name')
```
Just so you know, 'Coadd object base name' could be truncated in the header.

tbowers7 replied:
That's fine.  The comment is somewhat written in "most-significant-bit format" anyway, where any truncated words could be interpreted by the user.

debora-pe reviewed (approved):
Looks good. Thanks!

tbowers7 commented:
### DevSuite

Tests pass (`tbowers-kst-dec01.report`):
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  280 passed, 471 warnings in 385.96s (0:06:25) ---
--- PYTEST UNIT TESTS PASSED  158 passed, 93527 warnings in 903.35s (0:15:03) ---
--- PYTEST VET TESTS PASSED  69 passed, 143448 warnings in 4920.25s (1:22:00) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 284/284 TESTS  ---
Testing Started at 2025-12-01T19:24:29.978948
Testing Completed at 2025-12-02T09:26:03.777909
Total Time: 14:01:33.798961
```

tbowers7 committed:
PR Comment

	modified:   pypeit/history.py

kbwestfall commented on pypeit/history.py:
```diff
@@ -163,6 +163,29 @@ def add_coadd1d(self, spec1d_files, objids, gpm_exp=None):
+        self.append(f'PypeIt Coadded target {objname} in '
+                    f'{len(spec2d_files)} spec2d files')
+        for i, spec2d in enumerate(spec2d_files):
+            self.append(f'Exp {i}: "{Path(spec2d).name}"', add_date=False)
```
Nit-picking, but (correct me if I'm wrong) spec2d files can be made of multiple combined exposures.  So using "Exp" can be misleading.  Can we use "File" instead?

tbowers7 replied:
Yes, that's correct.  I was following the nomenclature in the logging... let me see about that...

tbowers7 replied:
@kbwestfall : Both the history and the logging have been changed from "exp" to "file".

kbwestfall reviewed (approved):
This looks great!  I had one nit-picky comment.  Let me know what you think, and then I'm read for this to be merged.

tbowers7 committed:
Nits

== picked.

References to "exp" have been replaced with "file" in the CoAdd2d history
and logging reports to represent the fact that input file to the coadding
can, themselves, be combined exposures (*e.g.*, from ABBA sequences).

	modified:   pypeit/coadd2d.py
	modified:   pypeit/history.py

debora-pe commented on pypeit/scripts/coadd_2dspec.py:
@tbowers7 Just want to mention that since you changed Exp to File, here there is another Exp that is printed in the log.

tbowers7 replied:
Thank you!  I missed that one in my search..replace.

tbowers7 committed:
Missed a nit

	modified:   pypeit/scripts/coadd_2dspec.py

tbowers7 committed:
Off-topic: Add info message when finding an okay STANDARD

	modified:   pypeit/core/standard.py

tbowers7 committed:
More nits (they keep multiplying)

Changed the output "exp -> file" wording, but there are MANY MANY more
examples of the work "exposure" in the comments and docstrings.

	modified:   pypeit/coadd2d.py

tbowers7 committed:
setup_coadd2d tweaks

* Remove spaces in target names (like elsewhere in PypeIt)
* When reading from a PypeIt file, only pull targets that have either
  'science' or 'standard' in the frametype.  This was written to also
  allow frames that are, e.g., 'arc,tilt' in addition to 'science'.

	modified:   pypeit/scripts/setup_coadd2d.py

tbowers7 commented:
@kbwestfall , see the "Late Additions" section of the PR description above when you go to re-review / merge.

tbowers7 committed:
Typos

	modified:   pypeit/coadd2d.py
	modified:   pypeit/scripts/setup_coadd2d.py

tbowers7 committed:
Merge branch 'develop' into objtrace_kst

kbwestfall reviewed (approved):
Looks great!  Thanks for the late additions.

kbwestfall committed:
Merge branch 'develop' into objtrace_kst
